File: python_programmers/level3/네트워크.py
def dfs(n,computers, idx, visited):
    # 방문 표시
    visited[idx] = True
    
    for i in range(n):
        if i != idx and computers[idx][i] == 1:
            if not visited[i]:
                dfs(n,computers,i, visited)
    

def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    for i in range(n):
        if not visited[i]:
            dfs(n, computers, i, visited)
            answer += 1
            
    return answer

File: javascript_programmers/level3/네트워크.js
const dfs = (computers, v, visited) => {
    // 방문 처리
    visited[v] = true;

    computers[v].forEach((computer, index) => {
        if(computer === 1 && visited[index] === false){
            dfs(computers, index, visited);
        }
    });

}

const solution = (n, computers) => {
    let networkCount = 0;
    const visited = Array(n).fill(false);

    visited.forEach((v, index) => {
        if(v === false){
            networkCount += 1;
            dfs(computers, index, visited);
        }
    });

    return networkCount;

}